qualtran/_infra/gate_with_registers.py

```diff
@@ -318,12 +319,42 @@ def controlled(
         control_qid_shape: Optional[Tuple[int, ...]] = None,
     ) -> 'cirq.Gate':
         # Multiple inheritance: use the `cirq.Gate` method.
-        return cirq.Gate.controlled(
+        from qualtran.cirq_interop import BloqAsCirqGate
+
+        controlled_gate = cirq.ControlledGate(
             self,
             num_controls=num_controls,
             control_values=control_values,
             control_qid_shape=control_qid_shape,
         )
+        ctrl_spec = CtrlSpec.from_cirq_cv(controlled_gate.control_values)
```

Comment:
So we make the gate so we can get the corresponding `CtrlSpec` (and throw out the gate). Is this a kludge or is this fine? If kludge: open an issue to track simplifying

Reply:
Eventually, I'd like to change the `def controlled` method to use the bloq style `def controlled(ctrl_spec)` instead of cirq style `def controlled(self, num_controls, control_values, ...)` so this simplification wouldn't be required.